active/0000-qunit-dom.md

```diff
@@ -0,0 +1,105 @@
+- Start Date: 2018-02-12
+- RFC PR: (leave this empty)
+
+# Summary
+
+Introduce [`qunit-dom`] as a dependency by default in the `app` and `addon` blueprints.
+
+[`qunit-dom`]: https://github.com/simplabs/qunit-dom
+
+# Motivation
+
+> Why are we doing this?
+
+QUnit assertions are quite verbose and can look like this:
+
+```js
+assert.equal(this.element.querySelector('.title').textContent.trim(), 'Hello World!');
+```
+
+with `qunit-dom` we can write much more readable assertion for DOM elements:
+
+```js
+assert.dom('.title').hasText('Hello World!');
+```
+
+> What use cases does it support?
+
+It supports the most common assertions on DOM elements, like:
+
+- what text does the element have?
+- what value does the `<input>` element have?
+- is a certain CSS class applied to the element
+
+The full API is documented at <https://github.com/simplabs/qunit-dom/blob/master/API.md>.
+
+> What is the expected outcome?
+
+Using `qunit-dom` will lead to more simple and readable test code.
+
+
+# Detailed design
+
+The necessary changes to `ember-cli` are relatively small since we only need
+to add the dependency to the `app` blueprint, and the `addon` blueprint will
+inherit it automatically.
```

Comment:
Shouldn't the test blueprints (part of `ember-source`) also be updated? E.g. [here](https://github.com/emberjs/ember.js/blob/master/blueprints/component-test/qunit-rfc-232-files/tests/__testType__/__path__/__test__.js#L24). This would also facilitate the discovery and learning process for users.

Reply:
Good point, I forgot about that... 🤔 

I'll update the text to add it